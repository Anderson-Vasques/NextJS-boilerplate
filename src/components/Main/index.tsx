const Main = () => <h1>foobar</h1>

export default Main
