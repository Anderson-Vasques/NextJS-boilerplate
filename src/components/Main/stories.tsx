import { withKnobs, text } from '@storybook/addon-knobs'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
}

export const Basic = () => (
  <Main
    title={text('title', 'React avanÃ§ado')}
    description={text(
      'description',
      'Typescript, ReactJS, NextJS e Styled Components'
    )}
  />
)
