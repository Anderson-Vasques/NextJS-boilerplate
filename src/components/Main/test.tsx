import { screen, render } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('Should render a heading', () => {
    const { container } = render(<Main />)

    expect(screen.getByRole('heading', { name: /foobar/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
