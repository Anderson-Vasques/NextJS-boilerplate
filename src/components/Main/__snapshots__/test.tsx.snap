// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render the heading 1`] = `
<main
  class="styles__Wrapper-sc-19t6oi2-0 kUWDBO"
>
  <img
    alt="Imagem de um átomo e React avançado escrito ao lado"
    class="styles__Logo-sc-19t6oi2-1 eCMFeh"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-19t6oi2-2 kIvHsX"
  >
    React avançado
  </h1>
  <h2
    class="styles__Description-sc-19t6oi2-3 kzKRNc"
  >
    Typescript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Desenvolvedo de frente para a tela com códdigo"
    class="styles__Ilustration-sc-19t6oi2-4 jyHLQR"
    src="/img/hero-illustration.svg"
  />
</main>
`
